homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,147 @@
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, value in sorted(SENSORS.items()):
+        sensors.append(KebaBinarrySensor(key, keba, value[0], value[1]))
+
+    async_add_entities(sensors)
+
+
+class KebaBinarrySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, key, keba, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Return False.
+
+        Data update is triggered from Keba entitiy.
+        """
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return "keba_" + self._name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return "keba_" + self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the binary sensor."""
+        return self._attributes.items()
+
+    async def async_update(self):
+        """Get latest cached states from the device."""
+        if self._key == 'Online':
+            self._state = self._keba.get_value(self._key)
+            self._attributes['Product'] = self._keba.get_value('Product')
+            self._attributes['Serial'] = self._keba.get_value('Serial')
+            self._attributes['Firmware'] = self._keba.get_value('Firmware')
+
+            secs = self._keba.get_value('Sec')
+
+            text = ""
+            if secs is not None:
+                if secs < 60:
+                    text = str(int(secs)) + " secs"
+                elif secs < 3600:
+                    text = str(int(secs / 60)) + " mins"
+                elif secs < 86400:
+                    text = str(int(secs / 3600)) + " hours"
+                elif secs < 31536000:
+                    text = str(int(secs / 86400)) + " days"
+                else:
+                    text = str(int(secs / 31536000)) + " years, " + \
+                           str(int((secs % 31536000) / 86400)) + " days"
+                self._attributes['Device uptime'] = text
+
+        elif self._key == 'Plug':
+            plug_state = self._keba.get_value(self._key)
+            if plug_state is not None:
+                self._state = plug_state > 3
+                self._attributes["Plugged on wallbox"] = plug_state > 0
+                self._attributes["Plug locked"] = \
+                    plug_state == 3 | plug_state == 7
+                self._attributes["Plugged on EV"] = plug_state > 4
```

Comment:
I would move handling of these magic values to the backend library, and create user-friendly getters for those:
```
@dataclass
class PlugState:
    state: int
    @property
    def is_plugged(self):
        return self.state > 0
    @property
    def is_locked(self):
        return self.state == 3 | self.state == 7
```
and so on.

Also, is binary or what you want for plug locked instead of `or`?